### ISSUE TITLE
EventID or correlationID missing in the logs with 4.3 upgrade

### ISSUE THREAD

**Issue by bhanupar** [mulesoft-consulting/json-logger#26]
4.2.2 log looks like

[01-11 06:45:37.824] INFO JsonLogger [[MuleRuntime].io.03: [test].testFlow.BLOCKING @6ce7b97d] [event: f26abdd5-1288-46e4-bc78-3b2cc027d069]: <<Log Message>>

4.3.0 is showing "Event ID is coming as NULL"

[01-11 04:50:18.409] INFO JsonLogger [[MuleRuntime].uber.34: [test2].testFlow.BLOCKING @c0b483] [event: ]:<<Log Message>>

**Comment by skuller** [comment]
@bhanupar  That event Id is not part of the info printed by the json logger. It's part of the default patternLayout from mule runtime.

**Comment by bhanupar** [comment]
Hi @skuller yes it’s part of pattern but with change in thread model UBER pool in 4.3 combination of Execution type “BLOCKING” and async call is having issues to have %X correlationid pattern in log4j which is passed between threads

**Comment by skuller** [comment]
@bhanupar for 4.3 I guess you could remove the blocking hint (https://github.com/mulesoft-consulting/json-logger/blob/mule-4.x/json-logger/src/main/java/org/mule/extension/jsonlogger/internal/JsonloggerOperations.java#L79) but it will be needed for anything lower than 4.3

**Comment by bhanupar** [comment]
@skuller  Thank you for the reply we also did  the same and it is working,
Sorry if this is not appropriate to ask here  I am a bit confused as per code the methods are async and by nature, these are unblocked and we explicitly made it BLOCKING, may I know if any reason behind it

**Comment by skuller** [comment]
@bhanupar the reason for that was that before mule runtime 4.3, non-blocking operations would go to the smaller thread pools. Since many customers use json logger to push data to third party systems (e.g. splunk) those HTTP calls created latency which would hold those threads for longer than they were intended to. Thus, engineering recommended forcing the logger to pick threads from the blocking pool which are more broadly available to avoid resource starvation. Hope this makes sense.

**Comment by bhanupar** [comment]
thank u so much, it really helps @skuller
